src/map.h

```diff
@@ -1052,6 +1067,20 @@ void add_corpse(int x, int y);
          * tr_null trap.
          */
         std::vector< std::vector<tripoint> > traplocs;
+
+    // cached values for map visibility calculations
+    int g_light_level;
```

Comment:
I broke up map::draw() because that was the only way to get as much code as possible de-duplicated. The code in cata_tiles that I was trying to eliminate is where I got the idea to embed and persist the values in the class. That's how it was done there. That is, without this patch, anyone playing with tiles is already using what you object to here. It might not be great, but it's no worse than what we already have for tiles players.

Calling the accessors 17k times per move instead of once was a big performance hit. Something like a 500ms delay per move in game when I tried it.